documentation/homework/week3/todo/src/App.js

```diff
@@ -0,0 +1,84 @@
+import React, { Component } from 'react';
+import './App.css';
+import moment from 'moment'
+
+// Components
+import AssignItem from './components/Assign'
+import ListItem from './components/ListItem'
+
+// Params
+import Params from './Params/locals'
+
+class App extends Component {
+  state = {
+    list: []
+  }
+  componentWillMount() {
+    this.setState({
+      list: Params.getLocals || []
+    })
+  }
+  Moments(date) {
+    const MOMENT_FORMAT = ["MM-DD-YYYY", "YYYY-MM-DD", "YYYY/MM/DD", "MM/DD/YYYY"]
+    return moment(date, MOMENT_FORMAT).format('LLLL')
+  }
+  handleRemove = (event, local_id) => {
+    this.setState({
+      list: Params.removeLocal(local_id)
+    })
+  }
+  handleAssign = (event) => { // Inserting Values
+    event.preventDefault()
+    const { deadline, description } = event.target
+    this.setState({
+      list: Params.saveLocal( // the Id is automaticly generats
```

Comment:
This is a pretty strange way to structure your updates. Inside `saveLocal`, you're actually writing data to `localStorage`, and then getting that data _back_ from localStorage and returning it. I would not recommend this, you should just set your React state directly, and then _respond_ to state changes inside lifecycle methods like `componentDidUpdate`.

In other words, you should just let React do its own thing. LocalStorage is only a "nice" thing we can add on top, it shouldn't be core functionality responsible for reading and writing state.

I would suggest you rewatch the video from Week 3, specifically how we implemented localStorage there. You have a bunch of localStorage functions like `removeLocal`, `updateLocal`, etc etc. These are all totally unnecessary and not good practice. You should be writing these functions inside your React component as handlers, not externally.

Then, you can just respond on each update cycle and write the state to localStorage, no matter what it is.